components/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountConnector.java

```diff
@@ -40,6 +40,19 @@
      */
     void deleteUserAccountAssociation(String userName) throws UserAccountAssociationException;
 
+    /**
+     * Delete an existing user account association of an owner's account
+     *
+     * @param ownerUserName Username of account trying to delete it's association.
+     * @param associatedUserName Username of account associated to delete associations of.
+     * @throws org.wso2.carbon.identity.user.account.association.exception.UserAccountAssociationException
+     */
+    default void deleteAssociatedUserAccount(String ownerUserName, String associatedUserName)
+            throws UserAccountAssociationException {
+
+        throw new UserAccountAssociationException("This is not implemented");
```

Comment:
Better to use one of the below options.

- Use the [NotImplementedException](http://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/NotImplementedException.html) from apache [commons-lang](http://commons.apache.org/proper/commons-lang/) which was available in commons-lang2 and has been re-added to commons-lang3 in version 3.2.
- Implement your own NotImplementedException.
- Throw an [UnsupportedOperationException](http://docs.oracle.com/javase/7/docs/api/java/lang/UnsupportedOperationException.html) with a message like "Not implemented, yet".

Reply:
Fixed in a8f731c7f18b25daffc0f805fdbe6755cf512c54